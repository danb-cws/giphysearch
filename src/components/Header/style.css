.App-header {
  font-size: 0.9rem;
  padding: 0.9em 2em 0 2em;
  color: var(--light-default);
  background-color: var(--primary-bg-color);
}
.App-header__title {
  margin: 0;
  font-size: 2em;
  font-weight: normal;
  text-transform: uppercase;
}
.App-header__intro {
  font-size: 1em;
  margin: 0.2em 0 0;
}
.App-welcome {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  position: relative;
  overflow: hidden;
  color: var(--secondary-color-inverse);
  background-color: var(--secondary-bg-color-inverse);
  font-size: 1rem;
}
.App-welcome__title {
  font-size: 8vw;
  font-weight: normal;
  text-transform: uppercase;
  letter-spacing: 0.2em;
  margin: 0;
}
.App-welcome__body {
  font-size: calc(7px + 2vw);
  max-width: 78vw;
  margin: 0.5em 0 0;
}
.github-link {
  display: inline-block;
  vertical-align: middle;
  margin-left: 0.85em;
  margin-top: -0.25em;
  text-indent: -999em;
  overflow: hidden;
  width: 1.5em;
  height: 1.5em;
  border: 1px solid var(--light-default);
  border-radius: 50%;
  background: var(--light-default) var(--github-logo-b64-url) no-repeat center
    center;
  background-size: 1.5em;
  transition: var(--default-transition);
}
.github-link:hover {
  transform: scale(1.3);
}
