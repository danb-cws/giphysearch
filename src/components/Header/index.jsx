import React from "react";
import "./style.css";

const Header = () => {
  return (
    <header className="App-header">
      <h1 className="App-header__title">Giphysearch</h1>
      <p className="App-header__intro">
        A React demo by Dan B
        <a
          href="https://github.com/danb-cws/giphysearch"
          className="github-link"
        >
          Github
        </a>
      </p>
    </header>
  );
};
export default Header;
