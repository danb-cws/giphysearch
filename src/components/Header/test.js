import React from "react";
import { shallow } from "enzyme";
import Header from "./";

describe("Header tests", () => {
  it("Renders and matches snapshot", () => {
    const tree = shallow(<Header />);
    expect(tree).toMatchSnapshot();
  });
  it("Shows a H1", () => {
    const el = shallow(<Header />);
    expect(
      el.contains(<h1 className="App-header__title">Giphysearch</h1>)
    ).toBe(true);
  });
});
