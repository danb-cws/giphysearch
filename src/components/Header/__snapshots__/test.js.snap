// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests Renders and matches snapshot 1`] = `
<Fragment>
  <header
    className="App-header"
  >
    <h1
      className="App-header__title"
    >
      Giphysearch
    </h1>
    <p
      className="App-header__intro"
    >
      A code demo by Dan B
      <a
        className="github-link"
        href="https://github.com/danb-cws/giphysearch"
        title="Github link"
      >
        Github
      </a>
    </p>
  </header>
  <section
    className="App-welcome"
  >
    <h2
      className="App-welcome__title"
    >
      Giphysearch
    </h2>
    <p
      className="App-welcome__body"
    >
      ...is a React application that allows you to search the Giphy API within a slick and snappy interface.
    </p>
    <p
      className="App-welcome__body"
    >
      To get started, enter a term in the search field.
    </p>
  </section>
</Fragment>
`;
